tests/__init__.py

```diff
@@ -359,3 +362,66 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class BotocoreHTTPStubber(object):
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._requests = []
+        self._responses = []
+
+    def add_response(self, response):
+        self._responses.append(response)
+
+    def create_response(self, url=None, status=200, headers=None, body=None):
+        if url is None:
+            url = 'https://example.com'
+
+        if headers is None:
+            headers = {}
+
+        if body is None:
+            body = b''
+
+        raw = RawResponse(body)
+        response = AWSResponse(url, status, headers, raw)
+        self._responses.append(response)
+
+    @property
+    def requests(self):
+        return self._requests
+
+    @property
+    def request_count(self):
+        return len(self._requests)
+
+    @contextlib.contextmanager
+    def wrap_client(self, client):
```

Comment:
What are your thoughts about exposing an `__enter__` and `__exit__` on the `HTTPStubber` that does this and have the `client` be passed into the `__init__()`?  It is more consistent to what is done in the `Stubber` class. Then you do not have to worry about passing in the client for every `with` statement:

```python
class TestSomething(unittest.TestCase):
    def setUp(self):
        self.client = session.create_client(...)
        self.http_stubber = HTTPStubber(self.client)


    def test_with_stubbed_layer(self):
        with self.http_stubber:
            self.http_stubber.add_response()
            self.client.make_call()
            # assert something
```